### PR TITLE
Added recipe for observability

### PR DESCRIPTION
**Pull request by NarmathaBala**
1. Changes to Tech Leads Checklist
2. Created new observability/readme.md for initial add

### CODE REVIEW

File: observability/readme.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## Goals
+
+1. Provide holistic view on the health of the application.
+2. Help measure business performance for the customer.
+3. Measure operational performance of the system.
+4. Identify and diagnose failures to get to the problem fast.
+
+## Recommended Practices
+
+1. Ensure health of the services are <b>monitored</b> and provide insights into system's performance and behavior.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
1. Ensure health of the services are **monitored** and provide insights into system's performance and behavior. 
```

File: observability/readme.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+1. Provide holistic view on the health of the application.
+2. Help measure business performance for the customer.
+3. Measure operational performance of the system.
+4. Identify and diagnose failures to get to the problem fast.
+
+## Recommended Practices
+
+1. Ensure health of the services are <b>monitored</b> and provide insights into system's performance and behavior. 
+2. <b>Faults, crashes and failures</b> are logged as discrete events. This helps engineers identify problem area(s) during failures.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
2. **Faults, crashes, and failures** are logged as discrete events. This helps engineers identify problem area(s) during failures.
```

File: observability/readme.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+3. Logging configuration (eg: verbose) can be controlled without code changes.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
3. Ensure logging configuration (eg: setting logging to "verbose") can be controlled without code changes.
```

File: observability/readme.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+4. Ensure that <b>metrics</b> around latency, duration are collected and are aggregatable.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
4. Ensure that **metrics** around latency and duration are collected and can be aggregated.
```

File: observability/readme.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+5. Include unique identifier for all logs to tie down aggregated data from various systems and provide a holistic view.
+6. Don't attempt to monitor everything. If the data is not actionable, it is useless and becomes noise.
+7. As a general rule, do not log any customer sensitive and Personal Identifiable Information (PII). Ensure GDPR compliance is followed regarding PII.
+8. Start small and add where there is customer impact.
+
+## What is collected
+
+1. Latency: Time taken to service a request.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
1. **Latency**: Time taken to service a request.
```

File: observability/readme.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+2. Traffic: Measurement of how much demand is placed on the system, typically number of requests per second.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
2. **Traffic**: Measurement of how much demand is placed on the system, typically number of requests per second.
```

File: observability/readme.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+3. Errors: Errors, application faults and traces with adequete information on error messages.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
3. **Errors**: Errors, application faults and traces with adequate information on error messages.
```

File: observability/readme.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+4. Identify a unique value (correlation identifier) that will be consistently tagged with data.
+
+In addition to above, engineers are encouraged to have conversations to understand other logs, metrics, data that adds value to the customer.
```
Comment (davidkpiano) [inline observability/readme.md]:
```suggestion
In addition to the above, engineers are encouraged to have conversations to understand other logs, metrics, data, etc. that adds value for the customer.
```

**Comment by NarmathaBala** [comment]
> Thanks for the contribution!

Thanks for the review, Fedrica

**Comment by NarmathaBala** [comment]
Fedrica, Ended up requested a second review by clumsy clicking :)

**Pull request merged**